src/gui/shareusergroupwidget.h

```diff
@@ -135,17 +139,33 @@ class ShareUserLine : public QWidget
 public slots:
     void slotStyleChanged();
 
+    void focusPasswordLineEdit();
+
 private slots:
     void on_deleteShareButton_clicked();
     void slotPermissionsChanged();
     void slotEditPermissionsChanged();
+    void slotPasswordCheckboxChanged();
     void slotDeleteAnimationFinished();
 
+    void refreshPasswordOptions();
+
+    void refreshPasswordLineEditPlaceholder();
+
+    void slotPasswordSet();
+    void slotPasswordSetError(int statusCode, const QString &message);
+
     void slotShareDeleted();
     void slotPermissionsSet();
 
     void slotAvatarLoaded(QImage avatar);
 
+    void setPasswordConfirmed();
+
+    void on_lineEdit_password_returnPressed();
```

Comment:
Please use explicit connects instead of the designer automagic. Otherwise if a change break them it won't be caught at compile time.